godot-macros/src/class/derive_godot_class.rs

```diff
@@ -287,7 +289,57 @@ fn make_user_class_impl(
         }
     };
 
-    let default_virtual_fn = if all_fields.iter().any(|field| field.is_onready) {
+    // Perform before-ready check only in debug mode.
+    // TODO - make sure if it shouldn't be hidden with another feature flag (for libraries running in editor, such as plugins).
+    #[cfg(debug_assertions)]
+    let oneditor_panic_inits = {
+        // Despite its name OnEditor shouldn't panic in the editor for tool classes.
+        let editor_check = quote! { ::godot::classes::Engine::singleton().is_editor_hint() };
+
+        // Inform the user which fields haven't been set, instead of panicking on the very first one. Useful for debugging.
+        let on_editor_fields_checks = all_fields
+            .iter()
+            .filter(|&field| field.is_oneditor)
+            .map(|field| {
+                let field = &field.name;
+                let warning_message =
+                    format! { "godot-rust: OnEditor field {field} hasn't been initialized."};
+                quote! {
+                if this.#field.is_invalid() {
+                    ::godot::global::godot_warn!(#warning_message);
+                    is_oneditor_properly_initialized = false;
+                }
+                }
+            })
+            .collect::<Vec<_>>();
+
+        if !on_editor_fields_checks.is_empty() {
+            run_before_ready = true;
+            quote! {
+                fn __are_oneditor_fields_initalized(this: &#class_name) -> bool {
+                    if #editor_check {
+                        return true;
+                    }
+
+                    let mut is_oneditor_properly_initialized: bool = true;
+                    #( #on_editor_fields_checks )*
+
+                    is_oneditor_properly_initialized
+                }
+
+                if !__are_oneditor_fields_initalized(&self) {
+                    panic!("godot-rust: OnEditor fields must be properly initialized before ready.")
+                }
+            }
+        } else {
+            TokenStream::new()
+        }
+    };
```

Comment:
This should maybe be its own function...

Reply:
I moved both `oneditor_panic_inits` and `onready_inits` to their own functions. Both return `TokenStreams` and the before_ready is applied if any of these TokenStreams isn't empty